sensorserial.ino

```diff
@@ -0,0 +1,722 @@
+	String label; // label to differentiate temperature readings
+};
+// Pins printed on the NodeMCU board are not the same as the esp8266 GPIO pin number, which is what we actually need to use here
+namespace NodeMcuPins
+{
+	enum : uint8_t
+	{
+		D0 = 16, D1 = 5, D2 = 4, D4 = 2,
+		D5 = 14, D6 = 12, D7 = 13, D8 = 15,
+		RX = 3, TX = 1, SD2 = 9, SD3 = 10
+	};
+}
+namespace Ads1115Pins
+{
+	enum : uint8_t
+	{
+		A0 = 0, A1 = 1, A2 = 2, A3 = 3
+	};
+}
+//----------------------------------------------------------
+// Change polling interval and sensor types here
+const unsigned long POLL_INTERVAL = 3000; // in milliseconds
+const SensorInfo SENSORS[] = 
+{// {sensor model, pin # (or I2C address for INA219)}
+	//{ina219, 0x40}, 
+	//{ina219, 0x41}, 
+	//{ads1115, Ads1115Pins::A0}, 
+	//{bme280, 0},
+	//{ds18b20, NodeMcuPins::SD3},
+	//{ads1115phototransistor, Ads1115Pins::A0},
+	//{ads1115anemometer, Ads1115Pins::A1},
+  {voltmeter,A0}
+};
+
+// --- DS18B20 Addresses ---
+// List sensor addresses here. Get them individually with ds18b20AddressFinder.ino or similar
+// Labels should be where the sensor is located or something. Or some identifying factor.
+const DS18B20Addresses DS18B20_ADDRESSES[] = {
+		{{0x28, 0x63, 0x1B, 0x49, 0xF6, 0xAE, 0x3C, 0x88}, "MCU"}, // measuring electronics enclosure temp,has white heatshrink
+		{{0x28, 0xDC, 0xA0, 0x49, 0xF6, 0xEB, 0x3C, 0xF0}, "Batt"}, // measuring battery enclosure temp, has white heatshrink
+		//{{0x28, 0x6A, 0x64, 0x49, 0xF6, 0xBE, 0x3C, 0xF0}, "red"} // has red heatshrink
+		};
+
+uint8_t NUM_SENSORS = sizeof(SENSORS) / sizeof(SENSORS[0]);
+uint8_t NUM_DS18B20S = sizeof(DS18B20_ADDRESSES) / sizeof(DS18B20_ADDRESSES[0]);
+
+//----------------------------------------------------------
+// Other Pin selection
+// Pins D5 (SCK), D6(MISO), D7(MOSI), and D8(CS) on NodeMCU are required for the SPI for the SD card reader, so don't use them.
+// On NodeMCU, pin D1(SCL) and D2(SDA) for I2C communication
+const uint8_t CS_PIN = NodeMcuPins::D8;// cs pin for sd card
+const uint8_t LED_PIN = LED_BUILTIN_AUX; // led pin for indicator. not required.
+
+// When adding new sensors, just make a wrapper that has the following variables and methods:
+class SensorWrap
+{
+	public:
+	uint8_t num_readings = 1; // how many readings to take from this sensor (a DHT22 has 2: temperature and humidity)
+	String labels; // label(s) for the data gathered from sensor
+	virtual void init(){} // whatever needs to go into setup()
+	virtual float getReading(uint8_t reading_num){return NAN;} // return measured value (NAN if no reading taken)
+};
+
+/* ----------------
+ * DS18B20 sensor: 
+ * Reads temperature. Many identical sensors can be wired to the same pin.
+ * num_readings does not need to be set manually. It's set in init() automatically.
+ * pin_num = OneWire data pin number
+ * If using an invalid address or there is a sensor missing, it will just read -196.6F
+ */
+class DS18B20Wrap : public SensorWrap
```

Comment:
Nearly all of this code is duplicated from the `sensorlogger.ino` file. I think it would be better to refactor this so both files are pulling classes from separate files